core/src/components/button/button.tsx

```diff
@@ -177,62 +202,58 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
       return form;
     }
     if (typeof form === 'string') {
+      // Check if the string provided is a form id.
       const el = document.getElementById(form);
       if (el instanceof HTMLFormElement) {
         return el;
       }
+      /**
+       * The developer specified a string for the form attribute, but the
+       * form element with that id does not exist in the DOM.
+       */
+      printIonWarning(
+        `Form with selector: "#${form}" could not be found. Verify that the id is correct and the form is rendered in the DOM.`,
```

Comment:
`el` could be either defined and not an instance of `HTMLFormElement` or not found in the DOM at this point. 

I'm thinking something like this, thoughts?

```ts
private findForm(): HTMLFormElement | null {
  const { form } = this;
  if (form instanceof HTMLFormElement) {
    return form;
  }
  if (typeof form === 'string') {
    // Check if the string provided is a form id.
    const el = document.getElementById(form);
    if (el) {
      if (el instanceof HTMLFormElement) {
        return el;
      } else {
        // warn about not an instance of HTMLFormElement
        return null;
      }
    } else {
      // warn about form id not found in document
      return null;
    }
  }
  /**
   * If the form element is not set, the button may be inside
   * of a form element. Query the closest form element to the button.
   */
  return this.el.closest('form');
}
```

Reply:
I like this. It provides more in-depth warnings. Just one question, are you proposing to not include the following [edge case](https://github.com/ionic-team/ionic-framework/pull/27790/files/83d04bd4ef7991c39095e0d0da1af529cbcc86e2#diff-8b0e185d2b3a4ceee4ec690a5f8ff2992ef3720594562f34bf2cae17f05254ffR220-R232)?

Reply:
Let's leave it in. I checked against the browser behavior here: https://codepen.io/sean-perkins/pen/rNQJozW

If you pass an "invalid" `form` value it does not submit the closest form.